version: '3.7'

services:
  fitdashboard-db:
    image: postgres:17.5
    env_file: .env
    ports:
      - 5432:5432
    networks:
      - fitdashboard-rede
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U ${POSTGRES_USER} -d ${POSTGRES_DB}"]
      interval: 5s
      timeout: 5s
      retries: 5


  django_fitdashboard:
    build: .
    command: sh -c "python manage.py migrate && python manage.py runserver 0.0.0.0:8000"
    env_file: .env
    environment:
      - DB_HOST=fitdashboard-db
      - DB_NAME=${POSTGRES_DB}
      - DB_PASSWORD=${POSTGRES_PASSWORD}
      - DB_USER=${POSTGRES_USER}
      - DB_PORT=${DB_PORT}

    ports:
      - 8000:8000
    volumes:
      - .:/FitDash
    depends_on:
      - fitdashboard-db
    networks:
      - fitdashboard-rede
    
    

networks:
  fitdashboard-rede:
    driver: bridge
